### PR TITLE
Backport EntrezGeneParser from ensembl-xref

### PR DESCRIPTION
**Pull request by mkszuba**
## Description

Port changes made to EntrezGeneParser in _ensembl-xref_, including unit tests, back to _ensembl:feature/xref_sprint_. Originally worked on by Alessandro.

## Use case

See ENSCORESW-3211.

## Benefits

Salvage more work from the 2018 xref sprint.

## Possible Drawbacks

Increased test-suite run time (note: Travis does not automatically execute tests from _misc-scripts/xref_mapping_ yet).

## Testing

_Have you added/modified unit tests to test the changes?_

Yes.

_If so, do the tests pass/fail?_

They will pass once #415 has been merged.

_Have you run the entire test suite and no regression was detected?_

Yes (including tests from _misc-scripts/xref_mapping_), no regression detected.

### CODE REVIEW

**Comment by mkszuba** [comment]
Comparison of output with _release/98_: identical.

**Comment by ens-bwalts** [comment]
Happy to defer the request to filter out `type_of_gene == 'other'` until a more reasonable time - you're correct about that change being beyond the scope of a backport (and, indeed, is a fairly major change that will need further investigation). Whether or not to uncompress the test data is also a minor issue (zcat exists) that I'm happy to defer to a later PR (if at all) as well. 

However, other parserers being backported do basic header validation, and it's a simple basic check that can prevent future headaches.

**Comment by mkszuba** [comment]
Uncompressed the test input file, added header validation, and while at it switched from (slow and not recommended even by Text::CSV authors) _getline_hr()_ and addressing columns by name to _getline()_ and addressing them by position.

File: misc-scripts/xref_mapping/XrefParser/EntrezGeneParser.pm [hunk 1]
```diff
@@ -18,6 +16,26 @@ limitations under the License.
...
+XrefParser::EntrezGeneParser
+
+=head1 DESCRIPTION
+
+This parser will read and create dependent xrefs from a simple
+comma-delimited file downloaded from the EntrezGene database.
+
+=head1 SYNOPSIS
+
+  my $parser = Bio::EnsEMBL::Xref::Parser::EntrezGeneParser->new($db->dbh);
```
Comment (tgrego) [inline misc-scripts/xref_mapping/XrefParser/EntrezGeneParser.pm]: old package name

Comment (mkszuba) [reply misc-scripts/xref_mapping/XrefParser/EntrezGeneParser.pm]: Oops. Fixed now.

**Pull request merged**